src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -0,0 +1,1278 @@
+	 *
+	 * Example
+	 *
+	 *     $p = WP_HTML_Processor::createFragment( '<template><strong><button><em><p><em>' );
+	 *     false === $p->next_tag();
+	 *     WP_HTML_Processor::ERROR_UNSUPPORTED === $p->get_last_error();
+	 *
+	 * @since 6.4.0
+	 *
+	 * @see self::ERROR_UNSUPPORTED
+	 * @see self::ERROR_EXCEEDED_MAX_BOOKMARKS
+	 *
+	 * @return string|null The last error, if one exists, otherwise null.
+	 */
+	public function get_last_error() {
+		return $this->last_error;
+	}
+
+	/**
+	 * Finds the next tag matching the $query.
+	 *
+	 * @TODO: Support matching the class name and tag name.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @throws WP_HTML_Unsupported_Exception
+	 *
+	 * @param array|string|null $query {
+	 *     Optional. Which tag name to find, having which class, etc. Default is to find any tag.
+	 *
+	 *     @type string|null $tag_name     Which tag to find, or `null` for "any tag."
+	 *     @type int|null    $match_offset Find the Nth tag matching all search criteria.
+	 *                                     1 for "first" tag, 3 for "third," etc.
+	 *                                     Defaults to first tag.
+	 *     @type string|null $class_name   Tag must contain this whole class name to match.
+	 *     @type string[]    $breadcrumbs  DOM sub-path at which element is found, e.g. `array( 'FIGURE', 'IMG' )`.
+	 * }
+	 * @return bool Whether a tag was matched.
+	 */
+	public function next_tag( $query = null ) {
+		if ( null === $query ) {
+			return $this->step();
+		}
+
+		if ( is_string( $query ) ) {
+			$query = array( 'breadcrumbs' => array( $query ) );
+		}
+
+		if ( ! is_array( $query ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				__( 'Please pass a query array to this function.' ),
+				'6.4.0'
+			);
+			return false;
+		}
+
+		if ( ! ( array_key_exists( 'breadcrumbs', $query ) && is_array( $query['breadcrumbs'] ) ) ) {
+			return $this->step();
+		}
+
+		if ( isset( $query['tag_closers'] ) && 'visit' === $query['tag_closers'] ) {
+			_doing_it_wrong(
+				__METHOD__,
+				__( 'Cannot visit tag closers in HTML Processor.' ),
+				'6.4.0'
+			);
+			return false;
+		}
+
+		$breadcrumbs  = $query['breadcrumbs'];
+		$match_offset = isset( $query['match_offset'] ) ? (int) $query['match_offset'] : 1;
+
+		$crumb  = end( $breadcrumbs );
+		$target = strtoupper( $crumb );
+		while ( $this->step() ) {
+			if ( $target !== $this->get_tag() ) {
+				continue;
+			}
+
+			// Look up the stack to see if the breadcrumbs match.
+			foreach ( $this->state->stack_of_open_elements->walk_up() as $node ) {
+				if ( strtoupper( $crumb ) !== $node->node_name ) {
+					break;
+				}
+
+				$crumb = prev( $breadcrumbs );
+				if ( false === $crumb && 0 === --$match_offset ) {
+					return true;
+				}
+			}
+
+			$crumb = end( $breadcrumbs );
+		}
+
+		return false;
+	}
+
+	/**
+	 * Steps through the HTML document and stop at the next tag, if any.
+	 *
+	 * ## Current HTML Support
+	 *
+	 *  - Doesn't support any HTML tags.
```

Comment:
What does this mean? At first glance, it reads to me like it cannot step through any HTML tags -- but that's basically any (meaningful) HTML, isn't it? Or does it mean it doesn't allow querying an HTML tag (like `next_tag` does)?